xtext-gradle-plugin/src/main/java/org/xtext/gradle/tasks/XtextEclipseSettings.xtend

```diff
@@ -5,15 +5,16 @@ import java.util.Set
 import org.eclipse.xtend.lib.annotations.Accessors
 import org.gradle.api.DefaultTask
 import org.gradle.api.JavaVersion
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.OutputFiles
 import org.gradle.api.tasks.TaskAction
 import org.xtext.gradle.protocol.GradleInstallDebugInfoRequest.SourceInstaller
 import org.xtext.gradle.tasks.internal.XtextEclipsePreferences
-import org.gradle.api.tasks.OutputFiles
 
 class XtextEclipseSettings extends DefaultTask {
 
-	@Accessors Set<XtextSourceDirectorySet> sourceSets
-	@Accessors Set<Language> languages
+	@Accessors @Internal Set<XtextSourceDirectorySet> sourceSets
```

Comment:
is this correct?

Reply:
Yes, but we'll also have to add an `upToDateWhen { false }` to the constructor, since this task doesn't track its inputs correctly. It can't do that even if we wanted to, because it actually mutates its own input files (reads preferences, then writes them back modified).

Reply:
which constructor?

Reply:
The constructor of the task.

Reply:
something like this

```
	new() {
		outputs.upToDateWhen [false]
	}
```

Reply:
Yep